src/simpleNbox.cpp

```diff
@@ -408,93 +433,110 @@ unitval SimpleNbox::getData(const std::string& varName,
                             const double date) throw ( h_exception )
 {
     unitval returnval;
+
+    // Does the varName contain our parse character? If so, split it
+    std::vector<std::string> splitvec;
+    boost::split( splitvec, varName, is_any_of( SNBOX_PARSECHAR ) );
+    H_ASSERT( splitvec.size() < 3, "max of one separator allowed in variable names" );
     
-    if( varName == D_ATMOSPHERIC_C ) {
+    std::string biome = SNBOX_DEFAULT_BIOME;
+    std::string varNameParsed = varName;
+    if( splitvec.size() == 2 ) {    // i.e., in form <biome>.<varname>
+        biome = splitvec[ 0 ];
+        varNameParsed = splitvec[ 1 ];
+    }
+
+    if( varNameParsed == D_ATMOSPHERIC_C ) {
         if(date == Core::undefinedIndex())
             returnval = atmos_c;
         else
             returnval = atmos_c_ts.get(date); 
-    } else if( varName == D_ATMOSPHERIC_CO2 ) {
+    } else if( varNameParsed == D_ATMOSPHERIC_CO2 ) {
         if(date == Core::undefinedIndex())
             returnval = Ca;
         else
             returnval = Ca_ts.get(date);
-    } else if( varName == D_ATMOSPHERIC_C_RESIDUAL ) {
+    } else if( varNameParsed == D_ATMOSPHERIC_C_RESIDUAL ) {
         if(date == Core::undefinedIndex())
             returnval = residual;
         else
             returnval = residual_ts.get(date);
-    } else if( varName == D_PREINDUSTRIAL_CO2 ) {
+    } else if( varNameParsed == D_PREINDUSTRIAL_CO2 ) {
         H_ASSERT( date == Core::undefinedIndex(), "Date not allowed for preindustrial CO2" );
         returnval = C0;
-    } else if(varName == D_BETA) {
-        // For the time being, we are only supporting global beta, not biome-specific
+    } else if(varNameParsed == D_BETA) {
         H_ASSERT(date == Core::undefinedIndex(), "Date not allowed for CO2 fertilization (beta)");
-        returnval = unitval(beta[SNBOX_DEFAULT_BIOME], U_UNITLESS);
-    } else if(varName == D_Q10_RH) {
+        returnval = unitval(beta[biome], U_UNITLESS);
```

Comment:
This is in `getdata()`, right?  So this means that if I send a message requesting a parameter for a biome that doesn't exist, an entry will be created under that biome, but _only_ for that parameter.  Is that going to create a problem.  Maybe it's harmless if nowhere else do we reference that biome, but if we ever infer the list of biomes by iterating over that parameter, then the calculation is going to turn into a charlie-foxtrot in a hurry.  

I think I've commented on this before.  If we're going to start using the code's biome capability, then we really need to be more intentional about creating biomes and keeping track of them.

Reply:
> So this means that if I send a message requesting a parameter for a biome that doesn't exist, an entry will be created under that biome, but only for that parameter.

This is true. 

>  Is that going to create a problem?

Almost definitely.

Will using the `.at` method alleviate this? I've seen comments elsewhere in the code that `.at(x)` will throw an error if `x` is not in the object, but I haven't checked.

Reply:
> then we really need to be more intentional about creating biomes and keeping track of them

Definitely agree. I mostly did this to have something working for my permafrost analysis. I think this PR can still be a useful starting point for discussions about how to proceed _properly_, even if this version of the code never sees the light of `master`.